lib/Monads/nondet/Nondet_Reader_Option.thy

```diff
@@ -70,6 +70,10 @@ lemma gets_the_obind:
   "gets_the (f |>> g) = gets_the f >>= (\<lambda>x. gets_the (g x))"
   by (rule ext) (simp add: monad_simps obind_def split: option.splits)
 
+lemma gets_the_Some_get:
```

Comment:
Maybe it would be a bit more "thematic" to state this with `ask`, rather than `Some`. I could then rename this to `gets_the_ask_get`, or simply `gets_the_ask` (I named this `gets_the_Some_get` only because `gets_the_Some` is already taken)

Reply:
I think I'm fine with `gets_the_Some_get` -- `ask` is mainly there for Haskell compatibility and is an input abbreviation only, i.e. when you see the term in your goal, you're going to see `Some`, not `ask`, and it'll be easier to find if the lemma has `Some` in its name.